File: Roadmap/01 - OPERADORES Y ESTRUCTURAS DE CONTROL/python/storbett.py


#operadore aritmeticos
print (f"suma: 10+3 = {10+3}")
print (f"resta: 10-3 = {10-3}")
print (f"mulriplicacion: 10*3 = {10*3}")
print (f"division: 10/3 = {10/3}")
print (f"modulo: 10%3 = {10%3}")
print (f"exponente: 10**3 = {10**3}")
print (f"division entera: 10//3 = {10//3}")

#operadores de comparacion

print (f"igualdad 10 == 3 es {10 == 3}")
print (f"desigualdad 10 != 3 es {10 != 3}")
print (f"mayor que 10 > 3 es {10 > 3}")
print (f"menor que 10 < 3 es {10 < 3}")
print (f"mayor o igual que 10 >= 3 es {10 > 3}")
print (f"menor o igual que 10 <= 3 es {10 < 3}")

#operadores logicos

print (f"AND &&: 10 + 3 == 13 and 5 -1 == 4 es {10 + 3 == 13 and 5 - 1 == 4}")
print (f"OR: 10 + 3 == 13 or 5 -1 == 4 es {10 + 3 == 13 or 5 - 1 == 4}")
print (f"NOT !: not 10 + 3 == 14 es {not 10 + 3 == 14}")


# operadores de asignacion 
my_number = 11 #asignacion
print (my_number)
my_number += 1
print (my_number)
my_number -= 1
print (my_number)
my_number *= 2
print (my_number)
my_number /= 2
print (my_number)
my_number %= 2
print (my_number)
my_number **= 1
print (my_number)
my_number //= 1

# operadores de identidad 
my_new_number = my_number
print (f"my_number is my_new_number es {my_number is my_new_number}")
print (f"my_number is not my_new_number es {my_number is not my_new_number}")

#operadores de pertenecia 

print (f" ' o ' in 'Simon' = {'o' in 'Simon'}")
print (f" ' h ' not in 'Simon' = {'o' not in 'Simon'}")

# operadores de bit

a = 10 # 1010
b = 3 # 0011
print (f"AND: 10 & 3 = {10 & 3}")
print (f"OR: 10 | 3 = {10 | 3}")
print (f"XOR: 10 ^ 3 = {10 ^ 3}")
print (f"NOT: 10 ~ 3 = {~10}")
print (f"desplazamiento a la derecha : 10 >> 2 = {10 >> 2}")
print (f"desplazamiento a la izquierda : 10 << 2 = {10 << 2}")


#estructuras de control

# Condicionales

my_string = "simons"
if my_string == "simons":
   print ("my_string es 'simons'")
elif my_string == "Brais":
   print ("my_string es brais")
else: 
   print ("my_string no es 'simons' ni 'brais'")

 # iterativas 

for i in range(11):
   print (i)

i = 0
while i <= 10:
   print (i)
   i+=1

# manejo de exepciones
try:
 print (10/0)
except:
   print ("se ha prodcuido un error")
finally:
   print ("ha finalizado el manejo de exepciones")


#extra

for number in range(10, 56):
   if number % 2 == 0 and number != 16 and number % 3 != 0:
      print (number)

File: Roadmap/01 - OPERADORES Y ESTRUCTURAS DE CONTROL/javascript/storbett.js
// Operadores aritméticos
console.log(`suma: 10 + 3 = ${10 + 3}`);
console.log(`resta: 10 - 3 = ${10 - 3}`);
console.log(`multiplicación: 10 * 3 = ${10 * 3}`);
console.log(`división: 10 / 3 = ${10 / 3}`);
console.log(`módulo: 10 % 3 = ${10 % 3}`);
console.log(`exponente: 10 ** 3 = ${10 ** 3}`);
console.log(`división entera: Math.floor(10 / 3) = ${Math.floor(10 / 3)}`);

// Operadores de comparación
console.log(`igualdad 10 == 3 es ${10 == 3}`);
console.log(`desigualdad 10 != 3 es ${10 != 3}`);
console.log(`mayor que 10 > 3 es ${10 > 3}`);
console.log(`menor que 10 < 3 es ${10 < 3}`);
console.log(`mayor o igual que 10 >= 3 es ${10 >= 3}`);
console.log(`menor o igual que 10 <= 3 es ${10 <= 3}`);

// Operadores lógicos
console.log(`AND &&: 10 + 3 == 13 && 5 - 1 == 4 es ${10 + 3 == 13 && 5 - 1 == 4}`);
console.log(`OR: 10 + 3 == 13 || 5 - 1 == 4 es ${10 + 3 == 13 || 5 - 1 == 4}`);
console.log(`NOT !: !(10 + 3 == 14) es ${!(10 + 3 == 14)}`);

// Operadores de asignación
let myNumber = 11; // asignación
console.log(myNumber);
myNumber += 1;
console.log(myNumber);
myNumber -= 1;
console.log(myNumber);
myNumber *= 2;
console.log(myNumber);
myNumber /= 2;
console.log(myNumber);
myNumber %= 2;
console.log(myNumber);
myNumber **= 1;
console.log(myNumber);
myNumber = Math.floor(myNumber / 1);
console.log(myNumber);

// Operadores de identidad
let myNewNumber = myNumber;
console.log(`myNumber === myNewNumber es ${myNumber === myNewNumber}`);
console.log(`myNumber !== myNewNumber es ${myNumber !== myNewNumber}`);

// Operadores de pertenencia
console.log(`'o' in 'Simon' = ${'Simon'.includes('o')}`);
console.log(`'h' not in 'Simon' = ${!('Simon'.includes('h'))}`);

// Operadores de bit
let a = 10; // 1010
let b = 3; // 0011
console.log(`AND: 10 & 3 = ${10 & 3}`);
console.log(`OR: 10 | 3 = ${10 | 3}`);
console.log(`XOR: 10 ^ 3 = ${10 ^ 3}`);
console.log(`NOT: ~10 = ${~10}`);
console.log(`desplazamiento a la derecha : 10 >> 2 = ${10 >> 2}`);
console.log(`desplazamiento a la izquierda : 10 << 2 = ${10 << 2}`);

// Estructuras de control

// Condicionales
let myString = "simons";
if (myString === "simons") {
    console.log("myString es 'simons'");
} else if (myString === "Brais") {
    console.log("myString es 'Brais'");
} else {
    console.log("myString no es 'simons' ni 'Brais'");
}

// Iterativas

for (let i = 0; i <= 10; i++) {
    console.log(i);
}

let i = 0;
while (i <= 10) {
    console.log(i);
    i++;
}

// Manejo de excepciones
try {
    console.log(10 / 0);
} catch (error) {
    console.log("se ha producido un error");
} finally {
    console.log("ha finalizado el manejo de excepciones");
}

// Extra
for (let number = 10; number < 56; number++) {
    if (number % 2 === 0 && number !== 16 && number % 3 !== 0) {
        console.log(number);
    }
}
